src/program.js

```diff
@@ -569,29 +569,31 @@ Example: $0 --help run.
           describe:
             'Use a proxy to access the signing API. ' +
             'Example: https://yourproxy:6000 ',
-          demandOption: false,
           type: 'string',
         },
         'use-submission-api': {
           describe:
             'Sign using the addon submission API',
-          demandOption: false,
           type: 'boolean',
         },
         'id': {
           describe:
             'A custom ID for the extension. This has no effect if the ' +
             'extension already declares an explicit ID in its manifest.',
-          demandOption: false,
           type: 'string',
         },
         'timeout': {
           describe: 'Number of milliseconds to wait before giving up',
           type: 'number',
         },
         'channel': {
-          describe: 'The channel for which to sign the addon. Either ' +
-          '\'listed\' or \'unlisted\'',
+          describe: 'The channel for which to sign the addon.',
+          type: 'string',
+          choices: ['listed', 'unlisted'],
+        },
+        'metadata': {
+          describe: 'A JSON object containing metadata to be passed to the ' +
```

Comment:
I think that for this option it may be useful if we could include in the doc string the url of the doc page where we describe what is the format for this metadata.